aws/resource_aws_rds_cluster_activity_stream.go

```diff
@@ -0,0 +1,255 @@
+	}
+
+	log.Printf("[DEBUG]: RDS Cluster start activity stream response: %s", resp)
+
+	d.SetId(resourceArn)
+
+	err = resourceAwsRDSClusterActivityStreamWaitForStarted(d.Timeout(schema.TimeoutCreate), d.Id(), conn)
+	if err != nil {
+		return err
+	}
+
+	return resourceAwsRDSClusterActivityStreamRead(d, meta)
+}
+
+func resourceAwsRDSClusterActivityStreamRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).rdsconn
+
+	input := &rds.DescribeDBClustersInput{
+		DBClusterIdentifier: aws.String(d.Id()),
+	}
+
+	log.Printf("[DEBUG] Describing RDS Cluster: %s", input)
+	resp, err := conn.DescribeDBClusters(input)
+
+	if isAWSErr(err, rds.ErrCodeDBClusterNotFoundFault, "") {
+		log.Printf("[WARN] RDS Cluster (%s) not found, removing from state", d.Id())
+		d.SetId("")
+		return nil
+	}
+
+	if err != nil {
+		return fmt.Errorf("error describing RDS Cluster (%s): %s", d.Id(), err)
+	}
+
+	if resp == nil {
+		return fmt.Errorf("error retrieving RDS cluster: empty response for: %s", input)
+	}
+
+	var dbc *rds.DBCluster
+	for _, c := range resp.DBClusters {
+		if aws.StringValue(c.DBClusterArn) == d.Id() {
+			dbc = c
+			break
+		}
+	}
+
+	if dbc == nil {
+		log.Printf("[WARN] RDS Cluster (%s) not found, removing from state", d.Id())
+		d.SetId("")
+		return nil
+	}
+
+	d.Set("arn", dbc.DBClusterArn)
+	d.Set("kms_key_id", dbc.ActivityStreamKmsKeyId)
+	d.Set("kinesis_stream_name", dbc.ActivityStreamKinesisStreamName)
+	d.Set("mode", dbc.ActivityStreamMode)
+
+	return nil
+}
+
+func resourceAwsRDSClusterActivityStreamUpdate(d *schema.ResourceData, meta interface{}) error {
+	if d.HasChanges("arn", "kms_key_id", "mode") {
+		log.Printf("[DEBUG] Stopping RDS Cluster Activity Stream before updating")
+		if err := resourceAwsRDSClusterActivityStreamDelete(d, meta); err != nil {
+			return err
+		}
+
+		log.Printf("[DEBUG] Starting RDS Cluster Activity Stream")
+		if err := resourceAwsRDSClusterActivityStreamCreate(d, meta); err != nil {
+			return err
+		}
+	}
+
+	return resourceAwsRDSClusterActivityStreamRead(d, meta)
+}
+
+func resourceAwsRDSClusterActivityStreamDelete(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).rdsconn
+
+	stopActivityStreamInput := &rds.StopActivityStreamInput{
+		ApplyImmediately: aws.Bool(true),
+		ResourceArn:      aws.String(d.Id()),
+	}
+
+	log.Printf("[DEBUG] RDS Cluster stop activity stream input: %s", stopActivityStreamInput)
+
+	resp, err := conn.StopActivityStream(stopActivityStreamInput)
+	if err != nil {
+		return fmt.Errorf("error stopping RDS Cluster Activity Stream: %s", err)
+	}
+
+	log.Printf("[DEBUG] RDS Cluster stop activity stream response: %s", resp)
+
+	if err := resourceAwsRDSClusterActivityStreamWaitForStopped(d.Timeout(schema.TimeoutDelete), d.Id(), conn); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func resourceAwsRDSClusterActivityStreamWaitForStarted(timeout time.Duration, id string, conn *rds.RDS) error {
```

Comment:
would be better to move all waiter funcs and logic to internal rds package alongside existing ones